pkg/controller/migplan/migplan_controller.go

```diff
@@ -254,6 +254,12 @@ func (r *ReconcileMigPlan) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	err = r.compareGVK(plan)
+	if err != nil {
+		log.Trace(err)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
```

Comment:
To be consistent with other validations and to keep `Reconcile()` clean, let's move this into:  `ReconcileMigPlan.validate()`.